### ISSUE TITLE
How to implement pause/resume behavior ?

### ISSUE THREAD

**Issue by lmindwarel** [atomashpolskiy/bt#146]
Hi

First of all, thanks for you work, very well done and documented.

I'm unable to implement a simple pause/pause mecanism, I haven't found the option to do this.

Currently, my start/resume function look like this: 
```
public void start() throws DownloadNotFoundException {
 
    if (client.isStarted()) {
      return;
    }

    try {
     client.startAsync(state -> {
       // doing something there
      }, 1000);
    } catch (Throwable e) {
      LOGGER.log(Level.INFO, "startAsync exception");
    }
  }
```

and my pause function like this:
```
public void pause(String id) throws DownloadNotFoundException {
    if (!client.isStarted()) {
      return;
    }

    try {
      client.stop();
    } catch (Throwable e) {
      LOGGER.log(Level.INFO, "stop exception");
    }
  }
```

Basically this is the same as the `CliClient.java` example.

But I have an exception when I try to resume a torrent: `Torrent already registered and data descriptor created`

So my question is: there is a way to resume a client without trying to register the torrent ?

Thank in advance
Antoine

**Comment by lmindwarel** [comment]
After some tests, the pause/resume mecanism doesn't seem to work either on `bt-cli`... maybe for the same reason.
I think the best workaround is to create a new client each time the download is started...

**Comment by atomashpolskiy** [comment]
Yeah, I recall there being some issues either with stop or pause, or both. Can't say precisely at the moment, sorry!

**Comment by wuge2019** [comment]
yeah

布吉岛
liucheng6699@vip.qq.com




---Original---

**Comment by lmindwarel** [comment]
No problem,

Maybe it might be a good idea not to throw an exception when torrent already registered and just ignore this case..

**Comment by atomashpolskiy** [comment]
Yeah, makes sense

**Comment by yjs112233** [comment]
> After some tests, the pause/resume mecanism doesn't seem to work either on `bt-cli`... maybe for the same reason.
> I think the best workaround is to create a new client each time the download is started...



> After some tests, the pause/resume mecanism doesn't seem to work either on `bt-cli`... maybe for the same reason.
> I think the best workaround is to create a new client each time the download is started...
Hi ! lmindwarel. 
I have the same problem as you. May I ask if "client" refers to Btclient or BtRuntime. This problem occurred when I downloaded the torrent again. Could you please share your solution.

**Comment by lmindwarel** [comment]
Hi,
I'm sorry I dont have a solution for doing this with this library..
I just use [ttorrent](https://github.com/mpetazzoni/ttorrent) instead :/

**Comment by atomashpolskiy** [comment]
As discussed earlier in this thread, TorrentRegistry has to be updated to not throw this exception.

**Comment by pyckle** [comment]
I think this was fixed in this commit: https://github.com/atomashpolskiy/bt/commit/1ad9d4caf49d89894712aa7f57085b88b2797505

Can this be closed?
